include/nlohmann/detail/macro_scope.hpp

```diff
@@ -4,13 +4,15 @@
 // You MUST include macro_unscope.hpp at the end of json.hpp to undef all of them
 
 // exclude unsupported compilers
-#if defined(__clang__)
-    #if (__clang_major__ * 10000 + __clang_minor__ * 100 + __clang_patchlevel__) < 30400
-        #error "unsupported Clang version - see https://github.com/nlohmann/json#supported-compilers"
-    #endif
-#elif defined(__GNUC__) && !(defined(__ICC) || defined(__INTEL_COMPILER))
-    #if (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__) < 40900
-        #error "unsupported GCC version - see https://github.com/nlohmann/json#supported-compilers"
+#ifndef _NLOHMANN_JSON_SKIP_COMPILER_CHECK_
```

Comment:
Symbols beginning with underscore followed by a capital letter are reserved by the standard.
You should remove the first and last underscores to be compliant.

(Applies to other macros added in this PR)

Reply:
Thanks, fixed that in the new commit.